src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -271,7 +292,9 @@ static ssize_t hwmon_sensor_show(struct device *dev,
 	char output[64];
 	uint32_t uval = 0;
 
+	mutex_lock(&sdm->sdm_lock);
 	get_sensors_data(sdm->pdev, repo_id);
+	mutex_unlock(&sdm->sdm_lock);
```

Comment:
I'm not sure how the locking schema is designed here, but after you drop the lock here, you still accessing sdm->sensor_data. Can it be changed now by someone else at the same time?

Reply:
yes, the sensor_data can be accessed by other thread/process. To fix it, driver shall copies the required data from sensor_data and then release the lock.

Reply:
The fix will make sure sysfs node entry points are multi-thread safe. But, when SDM runs in xclmgmt, there is one more entry point - sensor reading from xocl side. We need to also make sure that entry point is multi-thread safe and also make sure when both entry points are accessed at the same time, it is absolutely safe. I believe the locking schema needs to be designed and tested based on this.